dev-packages/application-manager/src/generator/backend-generator.ts

```diff
@@ -35,6 +35,13 @@ const { BackendApplication, CliManager } = require('@theia/core/lib/node');
 const { backendApplicationModule } = require('@theia/core/lib/node/backend-application-module');
 const { messagingBackendModule } = require('@theia/core/lib/node/messaging/messaging-backend-module');
 const { loggerBackendModule } = require('@theia/core/lib/node/logger-backend-module');
+const { BackendApplicationConfigProvider } = require('@theia/core/lib/node/backend-application-config-provider');
+
+const applicationConfigProvider = BackendApplicationConfigProvider.get();
```

Comment:
Could we define `ApplicationConfig` once in common module and use in the backend and frontend? With simple API like:
```
export namespace ApplicationConfig {
    export function set(params: ApplicationConfigParams) {} 
    export function get(): ApplicationConfigParams {}
}

Reply:
What is your concern with the current approach?

Reply:
It is verbose on client and implementations sides. I think it can be simpler.

Reply:
How can you distinguish between the frontend and the backend configuration types if you use the same API from common? If you know how to simplify this, just add your idea and I update this PR.

Reply:
at least, we could change that clients should not call `get().config` always, `applicationConfigProvider.get` could return `ApplicationConfig` directly

Reply:
I have followed the pattern we are currently using for the `ThemeService`. Why do we want to introduce a new construct? Yes, instead of returning the singleton, provider instance, I can return with the configuration, but the current way does not seem complicated to me either.

Reply:
ok.

I meant `ApplicationConfigProvider` seems to be a redundant thing just because `ApplicationConfig` is not a class. There can be namespace `ApplicationConfig` which reflects a static part of `ApplicationConfig` and adds `get/set` functions.

Also `ThemeService.get` returns `ThemeService`, not intermediate `ThemeServiceProvider`.

Reply:
> Also ThemeService.get returns ThemeService, not intermediate ThemeServiceProvider.

That is true, but we could have static methods there too without acquiring the instance first. I am OK changing it, in fact, let's change it.